src/service.c

```diff
@@ -156,6 +156,7 @@ static status_e activate_rpc( struct service *sp )
    socklen_t              sin_len = sizeof(tsin);
    unsigned long          vers ;
    struct service_config *scp = SVC_CONF( sp ) ;
+   uint16_t               service_port = SC_PORT( scp ) ;
```

Comment:
shouldn't this leverage `SC_SPECIFIED( scp, A_PORT )` ?  and set it to 0 if port isn't specified ?

Reply:
The usage here matches the existing logic in https://github.com/openSUSE/xinetd/blob/master/src/service.c#L236

I think both of these could probably be updated to use `SC_SPECIFIED( scp, A_PORT )` but that seems more appropriate as a separate followup change.

Reply:
you want to do that as another commit in this PR ? :)

Reply:
Here's a quick attempt.